does not work with ES 2.0.0-beta2

I think it crashes out parsing json from /_node/stats

```
[2015-10-07 10:05:39,451] DEBUG com.newrelic.metrics.publish.binding.Request - ComponentData(es-cluster:me.snov.newrelic-elasticsearch) : Metric: Component/V1/ClusterStats/Indices/Store/Size[bytes], count: 1, value: 3.10953344E9, minValue: 3.10953344E9, maxValue: 3.10953344E9, sumOfSquares: 9.669199E18
[2015-10-07 10:05:39,452] DEBUG com.newrelic.metrics.publish.Agent - Reporting metric: V1/ClusterStats/NumberOfVersionsInCluster
[2015-10-07 10:05:39,453] DEBUG com.newrelic.metrics.publish.binding.Request - ComponentData(es-cluster:me.snov.newrelic-elasticsearch) : Metric: Component/V1/ClusterStats/NumberOfVersionsInCluster[versions], count: 1, value: 1.0, minValue: 1.0, maxValue: 1.0, sumOfSquares: 1.0
SEVERE: An error has occurred
com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was NUMBER at line 1 column 2423 path $.nodes..os.load_average
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:200)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:103)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:196)
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:40)
	at com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:187)
	at com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:145)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:103)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:196)
	at com.google.gson.Gson.fromJson(Gson.java:810)
	at com.google.gson.Gson.fromJson(Gson.java:748)
	at me.snov.newrelic.elasticsearch.parsers.AbstractParser.parse(Unknown Source)
	at me.snov.newrelic.elasticsearch.parsers.AbstractParser.request(Unknown Source)
	at me.snov.newrelic.elasticsearch.ElasticsearchAgent.pollCycle(Unknown Source)
	at com.newrelic.metrics.publish.Runner$PollAgentsRunnable.run(Runner.java:209)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was NUMBER at line 1 column 2423 path $.nodes..os.load_average
	at com.google.gson.stream.JsonReader.beginArray(JsonReader.java:350)
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:79)
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:60)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:103)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:196)
	... 20 more
```

The os key from _node/stats looks like this 

```
os: {
    timestamp: 1444213509549,
    load_average: 0.11,
    mem: {
        total_in_bytes: 16659865600,
        free_in_bytes: 6217895936,
        used_in_bytes: 10441969664,
        free_percent: 37,
        used_percent: 63
    },
    swap: {
        total_in_bytes: 8371826688,
        free_in_bytes: 8371826688,
        used_in_bytes: 0
    }
}
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,10 +1,24 @@
+sudo: required
+dist: trusty
+
 language: java
 
+env:
+  - ELASTICSEARCH_VERSION="1.3"
+  - ELASTICSEARCH_VERSION="1.4"
+  - ELASTICSEARCH_VERSION="1.5"
+  - ELASTICSEARCH_VERSION="1.6"
+  - ELASTICSEARCH_VERSION="1.7"
+  - ELASTICSEARCH_VERSION="2.0"
+  - ELASTICSEARCH_VERSION="2.1"
+
 jdk:
-  - openjdk6
-  - openjdk7
   - oraclejdk7
   - oraclejdk8
+
+before_install:
+  - docker run -d -p 127.0.0.1:9200:9200 elasticsearch:$ELASTICSEARCH_VERSION
+  - sleep 5
 
 script:
   - ant
@@ -18,4 +32,6 @@ deploy:
     on:
         tags: true
         all_branches: true
-        condition: $TRAVIS_JDK_VERSION = openjdk6
+        condition:
+          - $TRAVIS_JDK_VERSION = oraclejdk7
+          - ELASTICSEARCH_VERSION = "2.1"

--- a/build.xml
+++ b/build.xml
@@ -2,7 +2,7 @@
 <project name="elasticsearch" default="dist" basedir="." xmlns:ivy="antlib:org.apache.ivy.ant">
 
     <!-- Change also me.snov.newrelic.elasticsearch.ElasticsearchAgent.VERSION -->
-    <property name="version" value="1.6.1" />
+    <property name="version" value="2.0.0" />
     <property name="filename" value="newrelic-elasticsearch-plugin-${version}"/>
     <property name="jarfile" value="plugin.jar" />
     <property name="tarfile" value="${filename}.tar.gz" />

--- a/src/me/snov/newrelic/elasticsearch/ElasticsearchAgent.java
+++ b/src/me/snov/newrelic/elasticsearch/ElasticsearchAgent.java
@@ -20,7 +20,7 @@ import java.net.MalformedURLException;
 public class ElasticsearchAgent extends Agent implements AgentInterface {
 
     private static final String GUID = "me.snov.newrelic-elasticsearch";
-    private static final String VERSION = "1.6.1";
+    private static final String VERSION = "2.0.0";
 
     private final String clusterName;
     private final ClusterStatsParser clusterStatsParser;

--- a/src/me/snov/newrelic/elasticsearch/reporters/NodesStatsReporter.java
+++ b/src/me/snov/newrelic/elasticsearch/reporters/NodesStatsReporter.java
@@ -152,20 +152,37 @@ public class NodesStatsReporter {
             reportNodeProcessedMetric("V1/NodeStats/ThreadPool/Index/Rejected", "threads/second", nodeName,
                     nodeStats.thread_pool.index.rejected);
 
-            // Merge
-            // Component/V1/NodeStats/ThreadPool/Merge/Completed/*
-            reportNodeProcessedMetric("V1/NodeStats/ThreadPool/Merge/Completed", "threads/second", nodeName,
-                    nodeStats.thread_pool.merge.completed);
+            if (nodeStats.thread_pool.force_merge != null) {
+                // Merge
+                // Component/V1/NodeStats/ThreadPool/Merge/Completed/*
+                reportNodeProcessedMetric("V1/NodeStats/ThreadPool/Merge/Completed", "threads/second", nodeName,
+                        nodeStats.thread_pool.force_merge.completed);
 
-            // Merge: queue
-            // Component/V1/NodeStats/ThreadPool/Merge/Queue/*
-            reportNodeMetric("V1/NodeStats/ThreadPool/Merge/Queue", "threads", nodeName,
-                    nodeStats.thread_pool.merge.queue);
+                // Merge: queue
+                // Component/V1/NodeStats/ThreadPool/Merge/Queue/*
+                reportNodeMetric("V1/NodeStats/ThreadPool/Merge/Queue", "threads", nodeName,
+                        nodeStats.thread_pool.force_merge.queue);
 
-            // Merge: rejected
-            // Component/V1/NodeStats/ThreadPool/Merge/Rejected/*
-            reportNodeProcessedMetric("V1/NodeStats/ThreadPool/Merge/Rejected", "threads/second", nodeName,
-                    nodeStats.thread_pool.merge.rejected);
+                // Merge: rejected
+                // Component/V1/NodeStats/ThreadPool/Merge/Rejected/*
+                reportNodeProcessedMetric("V1/NodeStats/ThreadPool/Merge/Rejected", "threads/second", nodeName,
+                        nodeStats.thread_pool.force_merge.rejected);
+            } else if (nodeStats.thread_pool.merge != null) {
+                // Merge
+                // Component/V1/NodeStats/ThreadPool/Merge/Completed/*
+                reportNodeProcessedMetric("V1/NodeStats/ThreadPool/Merge/Completed", "threads/second", nodeName,
+                        nodeStats.thread_pool.merge.completed);
+
+                // Merge: queue
+                // Component/V1/NodeStats/ThreadPool/Merge/Queue/*
+                reportNodeMetric("V1/NodeStats/ThreadPool/Merge/Queue", "threads", nodeName,
+                        nodeStats.thread_pool.merge.queue);
+
+                // Merge: rejected
+                // Component/V1/NodeStats/ThreadPool/Merge/Rejected/*
+                reportNodeProcessedMetric("V1/NodeStats/ThreadPool/Merge/Rejected", "threads/second", nodeName,
+                        nodeStats.thread_pool.merge.rejected);
+            }
 
             // Bulk
             // Component/V1/NodeStats/ThreadPool/Bulk/Completed/*
@@ -360,9 +377,8 @@ public class NodesStatsReporter {
 
         // Load average
         // Component/V1/NodeStats/Os/LoadAverage/*
-        if (nodeStats.os.load_average != null && nodeStats.os.load_average.size() > 0) {
-            reportNodeMetric("V1/NodeStats/Os/LoadAverage", "units", nodeName,
-                    nodeStats.os.load_average.get(0));
+        if (nodeStats.os.getLoadAverage() != null) {
+            reportNodeMetric("V1/NodeStats/Os/LoadAverage", "units", nodeName, nodeStats.os.getLoadAverage());
         }
 
         // Uptime

--- a/src/me/snov/newrelic/elasticsearch/responses/NodesStats.java
+++ b/src/me/snov/newrelic/elasticsearch/responses/NodesStats.java
@@ -1,6 +1,7 @@
 package me.snov.newrelic.elasticsearch.responses;
 
 import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 
 public class NodesStats {
@@ -152,10 +153,20 @@ public class NodesStats {
             }
 
             public Number uptime_in_millis;
-            public ArrayList<Double> load_average;
+            private Object load_average;
             public Cpu cpu;
             public Mem mem;
             public Swap swap;
+
+            public Number getLoadAverage() {
+                if (load_average instanceof List && ((List) load_average).size() > 0) {
+                    return ((List<Number>) load_average).get(0);
+                } else if (load_average instanceof Number) {
+                    return (Number) load_average;
+                } else {
+                    return null;
+                }
+            }
         }
         public static class Process {
             public static class Cpu {
@@ -244,6 +255,7 @@ public class NodesStats {
             public ThreadPoolStats bench;
             public ThreadPoolStats get;
             public ThreadPoolStats snapshot;
+            public ThreadPoolStats force_merge;
             public ThreadPoolStats merge;
             public ThreadPoolStats suggest;
             public ThreadPoolStats bulk;

--- a/test/me/snov/newrelic/elasticsearch/services/ClusterStatsParserTest.java
+++ b/test/me/snov/newrelic/elasticsearch/services/ClusterStatsParserTest.java
@@ -35,6 +35,14 @@ public class ClusterStatsParserTest {
     }
 
     @Test
+    public void testV210() throws Exception {
+        ClusterStats clusterStats = parseJson("/resources/cluster_stats_2.1.0.json");
+        assertEquals("elasticsearch", clusterStats.cluster_name);
+        assertEquals(1l, clusterStats.nodes.count.total.longValue());
+        assertEquals(1, clusterStats.nodes.versions.size());
+    }
+
+    @Test
     public void testV142() throws Exception {
         ClusterStats clusterStats = parseJson("/resources/cluster_stats_1.4.2.json");
         assertEquals("elasticsearch_snov", clusterStats.cluster_name);

--- a/test/me/snov/newrelic/elasticsearch/services/NodesStatsReporterTest.java
+++ b/test/me/snov/newrelic/elasticsearch/services/NodesStatsReporterTest.java
@@ -8,11 +8,14 @@ import org.junit.Test;
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.net.URL;
 
 import static org.junit.Assert.*;
 
 public class NodesStatsReporterTest {
 
+    private static final String nodesStatsUrl = "http://localhost:9200/_nodes/stats"; 
+    
     private MockAgent agent;
     private NodesStatsParser parser;
     private NodesStatsReporter reporter;
@@ -28,43 +31,56 @@ public class NodesStatsReporterTest {
         reporter = new NodesStatsReporter(agent);
     }
 
-    private NodesStats parseJson(String path) throws IOException {
-        InputStream stream = getClass().getResourceAsStream(path);
-        assertNotNull(String.format("Resource %s exists", path), stream);
+    private NodesStats parseJson(InputStream stream) throws IOException {
         return parser.parse(stream);
+    }
+    
+    private NodesStats parseJsonFromFile(String path) throws IOException {
+        return parseJson(getClass().getResourceAsStream(path));
+    }
+    
+    private NodesStats parseJsonFromUrl(String url) throws IOException {
+        return parseJson(new URL(url).openConnection().getInputStream());
     }
 
     @Test
     public void testReportNodesStatsV090() throws Exception {
-        NodesStats nodesStats = parseJson("/resources/nodes_stats_0.90.12.json");
+        NodesStats nodesStats = parseJsonFromFile("/resources/nodes_stats_0.90.12.json");
         reporter.reportNodesStats(nodesStats);
         assertTrue("Number of reported metrics > 0", agent.getReportedMetricsCount() > 0);
     }
 
     @Test
     public void testReportNodesStatsV134() throws Exception {
-        NodesStats nodesStats = parseJson("/resources/nodes_stats_1.3.4.json");
+        NodesStats nodesStats = parseJsonFromFile("/resources/nodes_stats_1.3.4.json");
         reporter.reportNodesStats(nodesStats);
         assertTrue("Number of reported metrics > 0", agent.getReportedMetricsCount() > 0);
     }
 
     @Test
     public void testReportNodesStatsV142() throws Exception {
-        NodesStats nodesStats = parseJson("/resources/nodes_stats_1.4.2.json");
+        NodesStats nodesStats = parseJsonFromFile("/resources/nodes_stats_1.4.2.json");
         reporter.reportNodesStats(nodesStats);
         assertTrue("Number of reported metrics > 0", agent.getReportedMetricsCount() > 0);
     }
 
     @Test
     public void testReportNodesStatsV142LimitedOsStats() throws Exception {
-        NodesStats nodesStats = parseJson("/resources/nodes_stats_1.4.2_incomplete_os_stats.json");
+        NodesStats nodesStats = parseJsonFromFile("/resources/nodes_stats_1.4.2_incomplete_os_stats.json");
         reporter.reportNodesStats(nodesStats);
         assertTrue("Number of reported metrics > 0", agent.getReportedMetricsCount() > 0);
     }
 
     @Test
     public void testReportNodesStatsV151() throws Exception {
-        NodesStats nodesStats = parseJson("/resources/nodes_stats_1.5.1.json");
+        NodesStats nodesStats = parseJsonFromFile("/resources/nodes_stats_1.5.1.json");
+        reporter.reportNodesStats(nodesStats);
+        assertTrue("Number of reported metrics > 0", agent.getReportedMetricsCount() > 0);
+    }
+
+    @Test
+    public void testReportNodesStatsV210() throws Exception {
+        NodesStats nodesStats = parseJsonFromUrl(nodesStatsUrl);
         reporter.reportNodesStats(nodesStats);
         assertTrue("Number of reported metrics > 0", agent.getReportedMetricsCount() > 0);
     }

--- a/test/resources/cluster_stats_2.1.0.json
+++ b/test/resources/cluster_stats_2.1.0.json
@@ -1,0 +1,113 @@
+{
+  "timestamp": 1450177715999,
+  "cluster_name": "elasticsearch",
+  "status": "green",
+  "indices": {
+    "count": 0,
+    "shards": {},
+    "docs": {
+      "count": 0,
+      "deleted": 0
+    },
+    "store": {
+      "size_in_bytes": 0,
+      "throttle_time_in_millis": 0
+    },
+    "fielddata": {
+      "memory_size_in_bytes": 0,
+      "evictions": 0
+    },
+    "query_cache": {
+      "memory_size_in_bytes": 0,
+      "total_count": 0,
+      "hit_count": 0,
+      "miss_count": 0,
+      "cache_size": 0,
+      "cache_count": 0,
+      "evictions": 0
+    },
+    "completion": {
+      "size_in_bytes": 0
+    },
+    "segments": {
+      "count": 0,
+      "memory_in_bytes": 0,
+      "terms_memory_in_bytes": 0,
+      "stored_fields_memory_in_bytes": 0,
+      "term_vectors_memory_in_bytes": 0,
+      "norms_memory_in_bytes": 0,
+      "doc_values_memory_in_bytes": 0,
+      "index_writer_memory_in_bytes": 0,
+      "index_writer_max_memory_in_bytes": 0,
+      "version_map_memory_in_bytes": 0,
+      "fixed_bit_set_memory_in_bytes": 0
+    },
+    "percolate": {
+      "total": 0,
+      "time_in_millis": 0,
+      "current": 0,
+      "memory_size_in_bytes": -1,
+      "memory_size": "-1b",
+      "queries": 0
+    }
+  },
+  "nodes": {
+    "count": {
+      "total": 1,
+      "master_only": 0,
+      "data_only": 0,
+      "master_data": 1,
+      "client": 0
+    },
+    "versions": [
+      "2.1.0"
+    ],
+    "os": {
+      "available_processors": 1,
+      "allocated_processors": 1,
+      "mem": {
+        "total_in_bytes": 0
+      },
+      "names": [
+        {
+          "name": "Linux",
+          "count": 1
+        }
+      ]
+    },
+    "process": {
+      "cpu": {
+        "percent": 0
+      },
+      "open_file_descriptors": {
+        "min": 95,
+        "max": 95,
+        "avg": 95
+      }
+    },
+    "jvm": {
+      "max_uptime_in_millis": 242552,
+      "versions": [
+        {
+          "version": "1.8.0_66-internal",
+          "vm_name": "OpenJDK 64-Bit Server VM",
+          "vm_version": "25.66-b17",
+          "vm_vendor": "Oracle Corporation",
+          "count": 1
+        }
+      ],
+      "mem": {
+        "heap_used_in_bytes": 53667240,
+        "heap_max_in_bytes": 1065025536
+      },
+      "threads": 22
+    },
+    "fs": {
+      "total_in_bytes": 19507089408,
+      "free_in_bytes": 17436667904,
+      "available_in_bytes": 16422166528,
+      "spins": "true"
+    },
+    "plugins": []
+  }
+}

